### PR TITLE
How to apply dataset filter expression while get parameter selection …

### PR DESCRIPTION
**Pull request by chille716**
…list? #1993

I added a boolean to enable/disable the normal data engine flow mode to apply filter expressions in datasets.

### CODE REVIEW

**Review by merks — COMMENTED** [review]
Wouldn't it be more general to have a getting and setting that returns a value of type DataEngineFlowMode?

**Comment by chille716** [comment]
@merks I was not sure about it... that's why I implemented it microinversively. Can't find any places where the other modes (IV, DATA_EXTRACTION) will be used.  But if it useful to set these modes at this place, then i can change it.

**Comment by merks** [comment]
FYI 

@speckyspooky  @wimjongman 

I don't really know the APIs well.  You're right that the other enumerators appear unused.  So I'll wait for other folks to comment.   Note that adding methods to API breaks the API so whatever we do here,  I think it would be better to define default methods in the interface, which means the mock class needs no changes....

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/api/IGetParameterDefinitionTask.java [hunk 1]
```diff
@@ -159,4 +159,20 @@ public interface IGetParameterDefinitionTask extends IEngineTask {
 	 */
 	Collection getSelectionTreeForCascadingGroup(String parameterGroupName);
 
+	/**
+	 * Enable or disable the normal flow mode. The normal flow mode can be used to
+	 * apply filters on data sets in data engine execution.
+	 *
+	 * @param normalFlowModeEnabled Set to true to enable the normal flow mode.
+	 */
+	void setNormalFlowModeEnabled(boolean normalFlowModeEnabled);
```
Comment (speckyspooky) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/api/IGetParameterDefinitionTask.java:168]: Add "default"-method for the new method - PR #1970, see IWordWriter.

grafik

File: engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/api/IGetParameterDefinitionTask.java [hunk 2]
```diff
@@ -159,4 +159,20 @@ public interface IGetParameterDefinitionTask extends IEngineTask {
...
+
+	/**
+	 * Is the normal flow mode enabled or not? If enabled, filters can be applied on
+	 * data sets.
+	 *
+	 * @return Returns true if the normal flow mode is enabled. Otherwise false.
+	 */
+	boolean isNormalFlowModeEnabled();
```
Comment (speckyspooky) [inline engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/api/IGetParameterDefinitionTask.java:176]: Add "default"-method for the new method - PR #1970, see IWordWriter.

**Review by speckyspooky — CHANGES_REQUESTED** [review]
Please add there default-methods to avoid broken API

**Review by speckyspooky — APPROVED** [review]
Looks good

**Pull request merged**